### ISSUE TITLE
Error when playing large twitch archives

### ISSUE THREAD

**Issue by kg** [streamlink/streamlink#471]
- [x] This is a bug report.

### Description

```'NoneType' object is not subscriptable``` error occurs when playing a large twitch video

### Expected / Actual behavior

A clearer playback failure reason would be good, alternately it could work. I'm guessing something is wrong with the HLS playlist for this archived stream.

### Reproduction steps / Stream URLs to test

1. https://www.twitch.tv/carcinogensda/v/116807345
Plays fine in the browser, but if played using streamlink (or livestreamer, fwiw) you get:
```
Traceback (most recent call last):
  File "C:\Program Files (x86)\Streamlink\bin\streamlink-script.py", line 12, in <module>
    main()
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink_cli\main.py", line 899, in main
    handle_url()
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink_cli\main.py", line 499, in handle_url
    handle_stream(plugin, streams, stream_name)
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink_cli\main.py", line 382, in handle_stream
    success = output_stream(stream)
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink_cli\main.py", line 249, in output_stream
    stream_fd, prebuffer = open_stream(stream)
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink_cli\main.py", line 225, in open_stream
    stream_fd = stream.open()
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink\stream\hls.py", line 261, in open
    reader.open()
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink\stream\segmented.py", line 185, in open
    self.worker = self.__worker__(self)
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink\stream\hls.py", line 128, in __init__
    self.reload_playlist()
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink\stream\hls.py", line 141, in reload_playlist
    playlist = hls_playlist.load(res.text, res.url)
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink\stream\hls_playlist.py", line 276, in load
    return parser(base_uri).parse(data)
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink\stream\hls_playlist.py", line 242, in parse
    self.parse_line(lineno, line)
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink\stream\hls_playlist.py", line 163, in parse_line
    segment = Segment(self.uri(line), extinf[0],
TypeError: 'NoneType' object is not subscriptable
```

### Environment details (operating system, python version, etc.)

Windows 10 x64, Python 2.7.11. Used the latest non-nightly Streamlink installer.

**Comment by beardypig** [comment]
Thanks for the detailed bug report, makes our lives so much easier :-)

You're right, it looks like a weird playlist. I'll take a look at the playlist parsing and see if there is a reasonable workaround. Probably the browser ignores invalid/unsupported lines in the playlist.

**Comment by beardypig** [comment]
Should be fixed in #473, twitch is producing invalid m3u8 playlists.

**Comment by gravyboat** [comment]
Closing since this is fixed, thanks again for the report!
